File: src/Services/ItemService.php
<?php

namespace App\Services;

use App\Models\Item;

class ItemService {
    private array $items;

    public function __construct() {
        $this->items = [
            new Item(1, "Car"),
            new Item(2, "Bike"),
            new Item(3, "House")
        ];
    }

    public function getAllItems(): array {
        return $this->items;
    }

    public function rentItem(int $id, string $startDate, string $endDate): ?Item {
        foreach ($this->items as $item) {
            if ($item->getId() === $id) {
                if ($item->rent($startDate, $endDate)) {
                    return $item;
                }
                throw new \Exception("Item is already rented or invalid rental period");
            }
        }
        throw new \Exception("Item not found");
    }

    public function getItemById(int $id): ?Item {
        foreach ($this->items as $item) {
            if ($item->getId() === $id) {
                return $item;
            }
        }
        throw new \Exception("Item not found");
    }
}

File: services/item_service.py
from models.item import Item

class ItemService:
    def __init__(self):
        self.items = [
            Item(1, "Car"),
            Item(2, "Bike"),
            Item(3, "House")
        ]

    def get_all_items(self) -> list:
        return self.items

    def rent_item(self, item_id: int, start_date: str, end_date: str) -> Item:
        item = next((i for i in self.items if i.id == item_id), None)
        if not item:
            raise Exception("Item not found")

        if not item.rent(start_date, end_date):
            raise Exception("Item is already rented or the period is invalid")
        
        return item

    def get_item_by_id(self, item_id: int) -> Item:
        item = next((i for i in self.items if i.id == item_id), None)
        if not item:
            raise Exception("Item not found")
        return item
